addons/mail/static/src/core/common/discuss_app_category_model.js

```diff
@@ -28,7 +28,22 @@ export class DiscussAppCategory extends Record {
     addTitle;
     /** @type {string} */
     addHotkey;
-    threads = Record.many("Thread");
+    threads = Record.many("Thread", {
+        sort(t1, t2) {
+            if (this.id === "channels") {
+                return String.prototype.localeCompare.call(t1.name, t2.name);
+            }
+            if (this.id === "chats") {
+                return t2.lastInterestDateTime.ts - t1.lastInterestDateTime.ts;
+            }
+            if (this.id === "livechat") {
+                return t2.lastInterestDateTime?.ts - t1.lastInterestDateTime?.ts;
+            }
+            if (this.id === "whatsapp") {
+                return t2.lastInterestDateTime.ts - t1.lastInterestDateTime.ts;
+            }
```

Comment:
`livechat` and `whatsapp` should not be referenced in `mail`

Also this sort method is not very well coded (already before the PR): when the 2 values are the same, it should check other criteria rather than returning 0, in particular `id` as last resort to make the sort deterministic.